test/specs/commonTests.js

```diff
@@ -0,0 +1,361 @@
+    _.each(Component._meta.props[propName], val => {
+      wrapper.should.not.have.className(val)
+    })
+  })
+}
+
+const _noClassNameFromBoolProps = (Component, propName, requiredProps) => {
+  const booleans = [true, false]
+
+  _.each(booleans, bool => it(`does not add any className when ${bool}`, () => {
+    // silence propType warnings
+    consoleUtil.disableOnce()
+
+    const wrapper = shallow(createElement(Component, { ...requiredProps, [propName]: bool }))
+
+    wrapper.should.not.have.className(propName)
+    wrapper.should.not.have.className('true')
+    wrapper.should.not.have.className('false')
+
+    _.each(Component._meta.props[propName], val => {
+      wrapper.should.not.have.className(val)
+    })
+  }))
+}
+
+const _classNamePropValueBeforePropName = (Component, propName, requiredProps) => {
+  it('adds "${value} ${name}" to className', () => {
+    const sample = _.sample(Component._meta.props[propName])
+    shallow(createElement(Component, { ...requiredProps, [propName]: sample }))
+      .should.have.className(`${sample} ${propName}`)
+  })
+}
+
+/**
+ * Assert that only a Component prop's name is converted to className.
+ * @param {React.Component|Function} Component The component to test.
+ * @param {String} propName A props key.
+ * @param {Object} requiredProps Props required to render the component.
+ */
+export const propKeyOnlyToClassName = (Component, propName, requiredProps = {}) => {
+  describe(propName, () => {
+    _noDefaultClassNameFromProp(Component, propName, requiredProps)
+
+    it(`adds prop name to className`, () => {
+      shallow(createElement(Component, { ...requiredProps, [propName]: true }))
+        .should.have.className(propName)
+    })
+
+    it(`does not add prop value to className`, () => {
+      // silence propType warnings
+      consoleUtil.disableOnce()
+
+      const value = faker.hacker.noun()
+      shallow(createElement(Component, { ...requiredProps, [propName]: value }))
+        .should.not.have.className(value)
+    })
+  })
+}
+
+/**
+ * Assert that only a Component prop's value is converted to className.
+ * @param {React.Component|Function} Component The component to test.
+ * @param {String} propName A props key.
+ * @param {Object} requiredProps Props required to render the component.
+ */
+export const propValueOnlyToClassName = (Component, propName, requiredProps = {}) => {
+  describe(propName, () => {
+    _definesPropOptions(Component, propName)
+    _noDefaultClassNameFromProp(Component, propName, requiredProps)
+    _noClassNameFromBoolProps(Component, propName, requiredProps)
+
+    it('adds prop value to className', () => {
+      const sample = _.sample(Component._meta.props[propName])
+      shallow(createElement(Component, { ...requiredProps, [propName]: sample }))
+        .should.have.className(sample)
+    })
+
+    it('does not add prop name to className', () => {
+      // silence propType warnings
+      consoleUtil.disableOnce()
```

Comment:
Note this util.  The test below will render this component with an intentionally incorrect prop value to assert that it does not have side effects.  This will throw big ugly propType warning logs in the middle of the tests.

This util let's us disable the console for one test.  It will automatically enable the console after this one test is complete.